AWMSmonitor.ino

```diff
@@ -0,0 +1,204 @@
+   LowPower library is used for putting arduino into sleep mode
+   Communication is done using radio wave transmission of packets.
+   LoRa modules are used for transmitting datas.
+*/
+#include <LowPower.h>
+#include <SPI.h>
+#include <LoRa.h>
+#include<String.h>
+#define ECHO_PIN 5            // Echo pin of Ultrasonic sensor
+#define TRIGGER_PIN 6         // Trigger pin of Ultrasonic sensor
+#define LORA_FREQUENCY 433E6  // Set LoRa frequency 
+#define BAUD_RATE 9600        // Serial baud rate
+#define onAck "mon"           // Expected Acknowledgement for motor power on (May be updated after cyphering)
+#define offAck "mof"          // Expected Acknowledgement for motor power off (May be updated after cyphering)
+#define ON_SLEEP 5            // Time in seconds for Board sleeps while motor is on
+#define OFF_SLEEP 900         // Time in seconds for Board sleeps while motor is off
+#define NO_ACK_SLEEP 60       // Time in seconds for Board sleeps while NACK state
+#define CS_PIN 10             // LoRa radio chip select
+#define RST_PIN 9             // LoRa radio Reset pin
+#define IRQ_PIN 2             // LoRa hardware interrupt pin
+/*
+ * Define DEBUG for Debug Mode
+ * DEBUG_LEVEL 0,1,2 can be used for various debugging levels
+ * 0: End user "ultra light mode"
+ * 1: End user "Normal mode"
+ * 2: Debug mode/Expert mode 
+*/
+#define DEBUG
+#define DEBUG_LEVEL 1
+#ifdef DEBUG
+#define debug_Println(level,msg) { \
+        if(DEBUG_LEVEL <= level) \
+        Serial.println(msg); \
+        }
+#define debug_Print(level,msg) { \
+        if(DEBUG_LEVEL <= level) \
+        Serial.print(msg); \
+        }
+#else
+#define debug_Println(x)
+#define debug_Print(x)
+#endif
+
+int sec = 1;                    //default sleep delay seconds
+byte localAddress = 0xFF;       // address of this device
+byte destination = 0x80;        // destination to send to
+String acknowledgment;
+long duration;
+float level;
+byte count = 0;
+String device_id = "IoTuneZ_906: "; // a fancy device id
+int loraval;
+bool isPktAvailable = false;
+String  pkt = "";
+bool isAcknoledged;
+
+void loraSetup()
+{
+  if (!LoRa.begin(LORA_FREQUENCY))
+    {
+      debug_Println(0,"Starting LoRa failed!");
+      while (1);
+    }
+}
+/*Function for arduino board to go to sleep for some seconds*/
+void system_sleep(int sleepVal)
+{
+  delay(5);
+  debug_Println(2,"going to sleep for " + String(sleepVal) + "seconds.");
+  Serial.flush();
+  for (sleepVal; (sleepVal) > 0; (sleepVal)--)
+  {
+    LowPower.idle(SLEEP_1S, ADC_OFF, TIMER2_OFF, TIMER1_OFF, TIMER0_OFF, SPI_OFF, USART0_OFF, TWI_OFF);
+    delay(5);
+  }
+  return;
+}
+void LoRa_sendMessage(String message)
+{
+  LoRa_txMode();                        // set tx mode
+  LoRa.beginPacket();                   // start packet
+  LoRa.print(message);                  // add payload
+  LoRa.endPacket(true);                 // finish packet and send it
+}
+
+void sendData()
+{
+  digitalWrite(TRIGGER_PIN, LOW);

```

Comment:
calculating distance and sending are two different functionalities
what is data means?

Reply:
ok, Data here is the level monitor data (read from the ultrasonic sensor) in a formatted way (including device ID, Message, 
 and Message ID)